File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    let total=0;
    let start=0;
    let end= 1000000000 * 100000
   
    while (start <= end) {
        middle = Math.floor((start + end) / 2);

        if (check(middle, n, times)) { // 범위를 더 줄여
            total = middle;
            end = middle - 1; // middle 왼쪽 범위
        } else {
            start = middle + 1; // middle 오른쪽 범위
        }
    }
    return total;
}

function check(curTime, n, times){
    let sum=0;
    
    for (const time of times) {
        sum += Math.floor(curTime / time);
    }

    if (sum < n) { // 주어진 인원수를 충족하지 못하는 경우
        return false;
    }
    
    return true; // 주어진 인원수를 충족하는 경우
}

File: 프로그래머스/3/43238. 입국심사/입국심사.java
class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        long left=1;
        long right = 1_000_000_000_000_000_000L;
        
        while(left<=right){
            long mid = (left + right) / 2;
            
            if(check(mid, n, times)) {
                answer=mid;
                right=mid-1;
            }
            else left=mid+1;
        }
        return answer;
    }
    public boolean check(long mid, int n, int[] times){
        long sum=0;
        
        for(int t : times) sum += mid/t;
        
        if(sum>=n) return true;
        return false;
    }
}